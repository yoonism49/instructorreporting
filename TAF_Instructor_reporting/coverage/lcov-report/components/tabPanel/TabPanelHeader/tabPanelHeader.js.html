<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/tabPanel/tabPanelHeader/TabPanelHeader.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">components/tabPanel/tabPanelHeader</a> TabPanelHeader.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">69.23% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/12</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">12x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&nbsp;
// @flow weak
&nbsp;
import React      from 'react';
import PropTypes  from 'prop-types';
import cx         from 'classnames';
&nbsp;
const TabPanelHeader = ({
  tabItems
}) =&gt; {
  const oneTabAtLeastIsActive = tabItems.some(item =&gt; item.isActive === true);
&nbsp;
  let tabItemsToDisplay = [...tabItems];
  <span class="missing-if-branch" title="if path not taken" >I</span>if (!oneTabAtLeastIsActive) {
<span class="cstat-no" title="statement not covered" >    tabItemsToDisplay = tabItems.map(<span class="fstat-no" title="function not covered" >(i</span>tem, itemIdx) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >      if (itemIdx === 0) {</span>
<span class="cstat-no" title="statement not covered" >        return {</span>
          ...item,
          isActive: true
        };
      }
<span class="cstat-no" title="statement not covered" >      return item;</span>
    });
  }
&nbsp;
  return (
    &lt;header className="panel-heading tab-bg-dark-navy-blue"&gt;
      &lt;ul className="nav nav-tabs"&gt;
        {
          tabItemsToDisplay.map(
            (item, itemIdx) =&gt; {
              const { name, tablink, isActive } = item;
&nbsp;
              return (
                &lt;li
                  key={itemIdx}
                  className={ cx({ active: isActive }) }&gt;
                  &lt;a
                    data-toggle="tab"
                    aria-expanded={isActive === true ? 'true' : 'false'}
                    href={`#${tablink}`}&gt;
                      &amp;nbsp;
                    {name}
                  &lt;/a&gt;
                &lt;/li&gt;
              );
            }
          )
        }
      &lt;/ul&gt;
    &lt;/header&gt;
  );
};
&nbsp;
TabPanelHeader.propTypes = {
  tabItems: PropTypes.arrayOf(
    PropTypes.shape({
      name:     PropTypes.string.isRequired,
      tablink:  PropTypes.string.isRequired,
      isActive: PropTypes.bool.isRequired
    }).isRequired
  ).isRequired
};
&nbsp;
export default TabPanelHeader;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Dec 04 2017 13:37:06 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
