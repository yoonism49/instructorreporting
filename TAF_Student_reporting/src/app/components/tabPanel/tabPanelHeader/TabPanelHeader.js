
// @flow weak

import React      from 'react';
import PropTypes  from 'prop-types';
import cx         from 'classnames';

const TabPanelHeader = ({
  tabItems
}) => {
  const oneTabAtLeastIsActive = tabItems.some(item => item.isActive === true);

  let tabItemsToDisplay = [...tabItems];
  if (!oneTabAtLeastIsActive) {
    tabItemsToDisplay = tabItems.map((item, itemIdx) => {
      if (itemIdx === 0) {
        return {
          ...item,
          isActive: true
        };
      }
      return item;
    });
  }

  return (
    <header className="panel-heading tab-bg-dark-navy-blue">
      <ul className="nav nav-tabs">
        {
          tabItemsToDisplay.map(
            (item, itemIdx) => {
              const { name, tablink, isActive } = item;

              return (
                <li
                  key={itemIdx}
                  className={ cx({ active: isActive }) }>
                  <a
                    data-toggle="tab"
                    aria-expanded={isActive === true ? 'true' : 'false'}
                    href={`#${tablink}`}>
                      &nbsp;
                    {name}
                  </a>
                </li>
              );
            }
          )
        }
      </ul>
    </header>
  );
};

TabPanelHeader.propTypes = {
  tabItems: PropTypes.arrayOf(
    PropTypes.shape({
      name:     PropTypes.string.isRequired,
      tablink:  PropTypes.string.isRequired,
      isActive: PropTypes.bool.isRequired
    }).isRequired
  ).isRequired
};

export default TabPanelHeader;
