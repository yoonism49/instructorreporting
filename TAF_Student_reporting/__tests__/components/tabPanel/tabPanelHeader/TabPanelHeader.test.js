
'use strict';

import React                from 'react';
import TabPanelHeader       from '../../../../src/app/components/tabPanel/tabPanelHeader/TabPanelHeader';
import renderer             from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('TabPanelHeader component', () => {
  it('renders as expected', () => {
    const mockProps = {
      tabItems: [
        {
          name: 'item1',
          tablink: '/item1',
          isActive: true
        },
        {
          name: 'item2',
          tablink: '/item2',
          isActive: false
        }
      ]
    };
    const component = renderer.create(
      <div>
        <TabPanelHeader { ...mockProps }>
          <p>TabPanelBody children here</p>
        </TabPanelHeader>
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
