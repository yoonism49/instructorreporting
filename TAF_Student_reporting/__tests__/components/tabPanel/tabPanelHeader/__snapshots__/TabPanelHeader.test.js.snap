// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanelHeader component renders as expected 1`] = `
<div>
  <header
    className="panel-heading tab-bg-dark-navy-blue"
  >
    <ul
      className="nav nav-tabs"
    >
      <li
        className="active"
      >
        <a
          aria-expanded="true"
          data-toggle="tab"
          href="#/item1"
        >
           
          item1
        </a>
      </li>
      <li
        className=""
      >
        <a
          aria-expanded="false"
          data-toggle="tab"
          href="#/item2"
        >
           
          item2
        </a>
      </li>
    </ul>
  </header>
</div>
`;
